src/trajectory.jl

```diff
@@ -227,22 +227,52 @@ samplecand(rng, τ::StaticTrajectory{EndPointTS}, h, z) = step(τ.integrator, h,
 
 ### Multinomial sampling from trajecory
 
-function randcat(rng::AbstractRNG, xs, p)
-    u = rand(rng)
+function randcat(rng::AbstractRNG, p_unorm::AbstractVector{T}) where {T}
+    p = p_unorm ./ sum(p_unorm)
+    u = rand(rng, T)
     cp = zero(eltype(p))
     i = 0
     while cp < u
-        cp += p[i +=1]
+        cp += p[i+=1]
     end
-    return xs[max(i, 1)]
+    return max(i, 1)
+end
+
+function randcat(rng, P_unorm::AbstractMatrix{T}) where {T}
```

Comment:
```suggestion
function randcat(rng::AbstractRNG, P_unorm::AbstractMatrix{T}) where {T}
```

Reply:
Yes but it would be `Union{AbstractRNG, AbstractVector{<:AbstractRNG}}`